lib/finch/http2/request_stream.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Finch.HTTP2.RequestStream do
+  @moduledoc false
+
+  defstruct [:body, :from, :from_pid, :status, :buffer, :continuation]
+
+  def new(body, {from_pid, _from_ref} = from) do
+    enumerable =
+      case body do
+        {:stream, stream} -> stream
+        data -> List.wrap(data)
+      end
+
+    enumerable =
+      Stream.map(enumerable, fn data ->
+        binary_data = IO.iodata_to_binary(data)
+        {binary_data, byte_size(binary_data)}
+      end)
```

Comment:
We could probably avoid calling `IO.iodata_to_binary/1` here if `data` is already a binary, right?